lib/sms.js

```diff
@@ -7,391 +7,390 @@ const _ = require('lodash');
-                if (params.retryDurationInHours) {
-                    body.retryDurationInHours = params.retryDurationInHours;
-                }
-            }
+    _validateParams();
 
-            const url = isBulk
-                ? Common.SMS_URL
-                : Common.CONTENT_URL + '/messaging';
+    // Multiple recipients?
+    if (validate.isArray(params.to)) {
+      if (params.to.length === 1) {
+        params.to = params.to[0];
+      } else {
+        params.to = params.to.join();
+      }
+    }
 
-            const rq = unirest.post(url);
-            rq.headers({
-                apikey: _self.options.apiKey,
-                Accept: _self.options.format
-            });
-            rq.send(body);
-            rq.end(function (resp) {
-                if (resp.status === 201) { // API returns CREATED on success!?
-                    resolve(resp.body);
-                } else {
-                    reject(resp.body || resp.error);
-                }
-            });
-        });
-    };
+    return new Promise(function (resolve, reject) {
+      if (validationError) {
+        return reject(validationError);
+      }
+
+      const body = {
+        username: _self.options.username,
+        to: params.to,
+        message: params.message,
+      };
+
+      if (params.from) {
+        body.from = params.from;
+      }
+
+      if (isBulk) {
+        body.bulkSMSMode = 1;
+        if (params.enqueue) {
+          body.enqueue = 1;
+        }
+        if (params.enqueue === false) {
+          body.enqueue = 0;
+        }
+      }
+
+      if (isPremium) {
+        body.bulkSMSMode = 0;
+        body.keyword = params.keyword;
+        body.linkId = params.linkId;
+        if (params.retryDurationInHours) {
+          body.retryDurationInHours = params.retryDurationInHours;
+        }
+      }
+
+      const url = isBulk ? Common.SMS_URL : Common.CONTENT_URL + '/messaging';
+
+      const rq = unirest.post(url);
+      rq.headers({
+        apikey: _self.options.apiKey,
+        Accept: _self.options.format,
+      });
+      rq.send(body);
+      rq.end(function (resp) {
+        if (resp.status === 201) {
+          // API returns CREATED on success!?
+          resolve(resp.body);
+        } else {
+          reject(resp.body || resp.error);
+        }
+      });
+    });
+  };
 }
 
-
 SMS.prototype.send = function (params) {
-    const opts = _.cloneDeep(params);
+  const opts = _.cloneDeep(params);
+
+  if (Array.isArray(opts)) {
+    const results = opts.map((opt) => {
+      return this._send(opt, true, false);
+    });
+
+    return Promise.allSettled(results).then((results) => {
+      return results.map((result, index) => {
+        if (result.status === 'fulfilled') {
+          return result.value;
+        } else {
+          return {
+            SMSMessageData: {
+              Message: `Not sent: ${result.reason.message}`,
+              recipients: opts[index].to,
```

Comment:
This assumes the `results` will have the same order as `opts`. Why not just use the result's reason as `Message` and not populate `Recipients`? i.e.
```
{
  SMSMessageData: {
       Message: result.reason,
       Status: 'Failed'
  }
}
```

Reply:
Noted. The above is actually a better format with the status of failed.